### PR TITLE
Setup MongoDB connection

### PR DESCRIPTION
**Pull request by jkwening**
## Issue Number:
#25 

## Issue Description:
Setup MongoDB connection

### Summary of solution:
1. Setup connection to local MongoDB in app.js
2. Add database.js to root/config/ with url to database connection port

#### Test

1. Run `npm start` at folder root - app should fail and print to console error noting connection failure and then exit the app
2. Standup local mongod database, run mongo cli, create 'app_db' database, and then run `npm start`, and confirm app is running successfully at localhost:3001 url
    - need 3 terminal instances: for running mongod database via `mongod --dbpath <path>` (default create data/db folder), for running mongo cli, and for running `npm start`

### Can this issue be closed?
Yes

### Should any new issues be added as a result of this solution?

- add folder for housing local mongo database. I used default root/data/db
- define database model
- setup API routes

### CODE REVIEW

**Comment by motosharpley** [comment]
Ok I see what your talking about Chris that does make sense I will notate the files as well

File: app.js [hunk 1]
```diff
@@ -1,8 +1,19 @@
 const bodyParser = require('body-parser');
 const express = require('express');
-
+const mongoose = require('mongoose');  // using ODM wrapper for MongoDB
+const database = require('./config/database');
```
Comment (motosharpley) [inline app.js]: to make the change Chris suggested we need to point this to config/index

File: app.js [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
 
+// set up MongoDB connection using the global promise library and then get connection
+mongoose.connect(database.url, {promiseLibrary: global.Promise}, error => {
+  if (error) {
+    console.log(`MongoDB connection error: ${error}`);
+    process.exit(1);  // should consider alternative to exiting the app due to db conn issue
+  }
+});
+const db = mongoose.connection;
+
```
Comment (motosharpley) [inline app.js]: the code here is fine just gonna throw this here as example of a different method cause it's interesting to see other soultions
 `// connect to mongoDB

mongoose.connect(database.url);
const db = mongoose.connection;

// Check connection
db.once('open', function(){
    console.log('Connected to MongoDB');
});

// Check for db errors
db.on('error', function(err){
    console.log(err);
});`

File: config/database.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    // TODO - replace 'app_db' with shorthand app name once that is decided
+    url : 'mongodb://localhost/app_db'  // using default port (27017)
+}
```
Comment (motosharpley) [inline config/database.js]: and then move this to dev.config.js
and we maybe throw a secret key on there just placeholder secret: 'yourSecret' or whatever

**Comment by jkwening** [comment]
@motosharpley updated PR has requested change - moving database url into dev/prod.config.js files. This should be good to go now..

**Pull request merged**